packages/caliper-core/lib/worker/workload/declarative/value-providers/value-provider-interface.js

```diff
@@ -0,0 +1,49 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+/**
+ * Interface for value providers
+ */
+class ValueProviderInterface {
+    /**
+     * Initialize an instance of ValueProviderInterface
+     * @param {object} options Options for configured parameter generation
+     * @param {object} variables Caliper provided variables
+     * @param {object} parameters User provided variables
+     * @param {object} valueProviderFactory ValueProviderFactory object reference
+     */
+    constructor(
+        options = {},
+        variables = {},
+        parameters = {},
+        valueProviderFactory
+    ) {
+        this.options = options;
+        this.variables = variables;
+        this.parameters = parameters;
+        if (valueProviderFactory === undefined) {
```

Comment:
This check should be left to inheriting classes, if they need the factory. We should remain flexible in the base class.